components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -2444,6 +2468,12 @@ void RewardsServiceImpl::SetBackupCompleted() {
   profile_->GetPrefs()->SetBoolean(prefs::kRewardsBackupSucceeded, true);
 }
 
+void RewardsServiceImpl::GetRewardsInternalsInfo(
+    const GetRewardsInternalsInfoCallback& callback) {
+  bat_ledger_->GetRewardsInternalsInfo(base::BindOnce(
+      &RewardsServiceImpl::OnGetRewardsInternalsInfo, AsWeakPtr(), callback));
```

Comment:
this actually doesn't need to use WeakPtr because `RewardsServiceImpl` will always outlive `bat_ledger_`, but it doesn't hurt anything. I just want to make sure everyone understands the issues with callbacks and object lifetimes